src/EnergyPlus/ThermalComfort.cc

```diff
@@ -1985,35 +1807,94 @@ namespace ThermalComfort {
 		// Locals
 		Real64 SurfaceTemp;
 
-		// SUBROUTINE ARGUMENT DEFINITIONS:
 		// SUBROUTINE PARAMETER DEFINITIONS:
-		// INTERFACE BLOCK SPECIFICATIONS:
-		// na
-
-		// DERIVED TYPE DEFINITIONS:
-		// na
+		Real64 const KelvinConv( 273.15 ); // Conversion from Celsius to Kelvin
 
 		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
 		int SurfNum;
-		Real64 SurfTempAngleFacSummed;
-
-		// FLOW:
+		Real64 SurfTempEmissAngleFacSummed;
 
-		SurfTempAngleFacSummed = 0.0;
-
-		for ( SurfNum = 1; SurfNum <= AngleFactorList( AngleFacNum ).TotAngleFacSurfaces; ++SurfNum ) {
-
-			SurfaceTemp = TH( 2, 1, AngleFactorList( AngleFacNum ).SurfacePtr( SurfNum ) );
-			SurfTempAngleFacSummed += SurfaceTemp * AngleFactorList( AngleFacNum ).AngleFactor( SurfNum );
+		SurfTempEmissAngleFacSummed = 0.0;
 
+		auto & thisAngFacList( AngleFactorList( AngleFacNum ) );
+		
+		for ( SurfNum = 1; SurfNum <= thisAngFacList.TotAngleFacSurfaces; ++SurfNum ) {
+			SurfaceTemp = TH( 2, 1, thisAngFacList.SurfacePtr( SurfNum ) ) + KelvinConv;
+			SurfTempEmissAngleFacSummed += thisAngFacList.SurfaceEmissivity( SurfNum ) * thisAngFacList.AngleFactor( SurfNum ) * pow_4( SurfaceTemp );
 		}
 
-		CalcAngleFactorMRT = SurfTempAngleFacSummed;
+		CalcAngleFactorMRT = root_4(SurfTempEmissAngleFacSummed/thisAngFacList.SumSurfaceEmissAngleFactor) - KelvinConv;
 
 		return CalcAngleFactorMRT;
 
 	}
 
+	Real64
+	CalcSurfaceWeightedMRT(
+		int const ZoneNum,
+		int const SurfNum
+	)
+	{
+		
+		// Purpose: Calculate a modified zone MRT that excludes the Surface( SurfNum ).
+		//          This is necessary for the surface weighted option to not in essence
+		//          double count SurfNum in the MRT calculation.  Other than that, the
+		//          method here is the same as CalculateZoneMRT.
+		
+		// Using/Aliasing
+		using DataHeatBalSurface::TH;
+		using DataSurfaces::Surface;
+		using DataSurfaces::TotSurfaces;
+		using DataHeatBalance::Construct;
+
+		// Return value
+		Real64 CalcSurfaceWeightedMRT;
+		
+		// Local variables
+		int SurfNum2; // surface number used in "for" loop
+		int ZoneNum2; // zone number index
+		Real64 SumAET; // Intermediate calculational variable (area*emissivity*T) sum
+		static Array1D< Real64 > SurfaceAE; // Product of area and emissivity for each surface
+		static Array1D< Real64 > ZoneAESum; // Sum of area times emissivity for all zone surfaces
+
+		// Initialize ZoneAESum for all zones and SurfaceAE for all surfaces
+		if ( FirstTimeSurfaceWeightedFlag ) {
+			FirstTimeSurfaceWeightedFlag = false;
+			SurfaceAE.allocate( TotSurfaces );
+			ZoneAESum.allocate( NumOfZones );
+			SurfaceAE = 0.0;
+			ZoneAESum = 0.0;
+			for ( SurfNum2 = 1; SurfNum2 <= TotSurfaces; ++SurfNum2 ) {
+				if ( Surface( SurfNum2 ).HeatTransSurf ) {
+					SurfaceAE( SurfNum2 ) = Surface( SurfNum2 ).Area * Construct( Surface( SurfNum2 ).Construction ).InsideAbsorpThermal;
```

Comment:
Again, this can't be precalculated if EMS actuates the thermal absorptance.  Since this is acting on a single zone now, it should simple enough to just have one loop that calculates both ZoneAESum and SumAET, exlcuding the weighted surface.

Reply:
@mjwitte Got it.  Will fix this also.